java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -18,45 +19,46 @@ interface UnleashFFI extends Library {
 
     UnleashFFI INSTANCE = Native.load(combinedPath, UnleashFFI.class);
 
-    Pointer engine_new();
+    Pointer new_engine();
 
     void engine_free(Pointer ptr);
 
-    Pointer engine_take_state(Pointer ptr, String toggles);
+    Pointer take_state(Pointer ptr, String toggles);
 
-    boolean engine_is_enabled(Pointer ptr, String name, String context);
+    boolean check_enabled(Pointer ptr, String name, String context);
 
-    Pointer engine_get_variant(Pointer ptr, String name, String context);
+    Pointer check_variant(Pointer ptr, String name, String context);
 
-    void engine_free_variant_def(Pointer variant_def_ptr);
+    void free_response(Pointer pointer);
 }
 
 public class UnleashEngine {
     private Pointer ptr;
     private ObjectMapper mapper = new ObjectMapper();
 
     public UnleashEngine() {
-        ptr = UnleashFFI.INSTANCE.engine_new();
+        ptr = UnleashFFI.INSTANCE.new_engine();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
This is just for being resilient when adding new fields that come from the engine state

Reply:
Nice!